src/NeoSharp.Core/Network/Server.cs

```diff
@@ -26,48 +28,40 @@ public class Server : IServer, IDisposable
         // if we can't connect to a peer it is inserted into this list
         // ReSharper disable once NotAccessedField.Local
         private readonly IList<IPEndPoint> _failedPeers;
-        private readonly EndPoint[] _peerEndPoints;
         #endregion
 
         #region Constructor
 
         /// <summary>
         /// Constructor
         /// </summary>
-        /// <param name="config">Network configuration</param>
         /// <param name="peerFactory">Factory to create peers from endpoints</param>
         /// <param name="peerListener">Listener to accept peer connections</param>
         /// <param name="aclLoader">ACL loader to define access</param>
         /// <param name="logger">Logger</param>
         /// <param name="serverContext">Server context</param>
         /// <param name="peerMessageListener">PeerMessageListener</param>
         public Server(
-            NetworkConfig config,
             IPeerFactory peerFactory,
             IPeerListener peerListener,
             INetworkAclLoader aclLoader,
             ILogger<Server> logger,
             IServerContext serverContext,
             IPeerMessageListener peerMessageListener)
         {
-            if (config == null) throw new ArgumentNullException(nameof(config));
-            _peerFactory = peerFactory ?? throw new ArgumentNullException(nameof(peerFactory));
-            _peerListener = peerListener ?? throw new ArgumentNullException(nameof(peerListener));
+            _peerFactory = Guard.ThrowIfNull(peerFactory, nameof(peerFactory));
```

Comment:
@shargon is against Guard pattern
I am against as well
@gubanotorious is neutral
@aboimpinto is up to it

Do we need more votes to decide?